### DISCUSSION TITLE
Error creating aab for Android

### DISCUSSION THREAD

**Discussion by idanasher** [airsdk/Adobe-Runtime-Support#1780] [Q&A]
I keep getting errors on my windows animate while trying to publish as aab.
Have tried so many "combinations" & tricks, nothing works and this issue is really messy.

on my  adt.cfg :
CreateAndroidAppBundle=true
AndroidPlatformSDK=C:/Users/Edan/AppData/Local/Android/Sdk
JAVA_HOME=C:/Program Files/Android/Android Studio1/jre

on my system enviroment varailbes:
JAVA_HOME=C:\Program Files\Android\Android Studio1\jre

on cmd java -version:
openjdk version "11.0.10" 2021-01-19
OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
OpenJDK 64-Bit Server VM (build 11.0.10+0-b96-7249189, mixed mode)

my sdk build-tools/l
![image](https://user-images.githubusercontent.com/8189469/158546862-84db63f5-f4c5-4f31-aecf-12f1043dea35.png)

when all is hided except 30.0.3, I get this error:
![image](https://user-images.githubusercontent.com/8189469/158547188-3559e580-c6ff-4215-a8eb-4d12c90eac7e.png)

when I "unhide" and have all folders visible, I get this error:
![image](https://user-images.githubusercontent.com/8189469/158547140-5bbd6747-05ae-4411-97aa-df873a4afb85.png)

Tried to publish with AIR 33.1.1.795 & AIR 33.1.1.633

Need your help

**Comment by marchbold** [comment]
This is not a bug in the AIR SDK, but an issue with your manifest additions. 

The error in the output there:

```
Error: android:exported needs to be explicitly specified for <activity>. Apps targeting Android 12 ... 
```

You need to update them for Android 31 or set your target sdk to 30 (currently recommended) .
